### ISSUE TITLE
Neural net not working

### ISSUE THREAD

**Issue by weiserhase** [keras-team/keras#14001]
**System information**  
- Have I written custom code (as opposed to using example directory):  yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  Windows 10
- TensorFlow backend (yes / no):  yes
- TensorFlow version:  2.3.1
- Keras version:  2.3.1
- Python version:  3.8.2

**Describe the current behavior**  
Imn trying to predict a cars MPG by its horsepower i get the data from a json file the data is imported and given to the neuralNet but the neural net just returns a straight line as prediction its simply 0.
The Neural net is also showing up corectly in nodel.summary().
 I think this is a issue with my code but im not getting the problem .

`import keras as k
import keras
from keras.models import Sequential
from keras.optimizers import SGD
from keras.layers import Dense, Activation, Dropout
import numpy as np
import matplotlib.pyplot as plt 
from random import randint
import json
x = list()
y = list()
def f(x):
    y = x*randint(0,10)
    return y 
with open('r.json') as myfile:
    data=myfile.read()
data = json.loads(data)
# parse file
x = []
y = []
for i in range(len(data)):
    #car = json.loads(data[i])
    car = data[i]
    if(car["Miles_per_Gallon"] is'None' or car["Miles_per_Gallon"] is None):
        continue
    if(car["Horsepower"] is None or car["Horsepower"] is 'null' or car["Cylinders"]  is None or car['Acceleration'] is None or car['Weight_in_lbs'] is None ):
        continue
    if(car["Horsepower"] != None or car["Horsepower"] != 'null' or car["Cylinders"] !=None or car['Acceleration'] != None or car['Weight_in_lbs'] != None ):
        x.append([ car["Horsepower"]])
        if(car["Miles_per_Gallon"] != 'None' or car["Miles_per_Gallon"] != None):
            print(car )
            y.append(car["Miles_per_Gallon"])
data = x
print((x))
print((y))
model = Sequential()
model.add(Dense(1, input_dim=1, use_bias=False, activation='sigmoid'))
model.add(Dense(100, activation='tanh',))
model.add(Dense(50, activation='tanh',))
model.add(Dense(25, activation='tanh',))
model.add(Dense(25, activation='tanh',))
model.add(Dense(25, activation='tanh',))
model.add(Dense(25, activation='tanh',))
model.add(Dense(1, activation='relu',))
sgd = SGD(lr=0.01, momentum=0.9, nesterov=True)
model.compile( loss = "mean_squared_error", 
               optimizer = sgd, 
               metrics=['accuracy']
             )

weights = model.layers[0].get_weights()
print(weights)

model.fit(np.array(x), np.array(y),
          batch_size=int(len(x)*0.9),epochs=5000,
          )

weights = model.layers[0].get_weights()
print(weights)

predict = model.predict([data])

for i in range (10):
    t = randint(0,len(data))
    print(y[t])
    testData = x[i]
    print(model.predict( np.array([testData])))
    
plt.plot(data, predict, 'b', data , y, 'k.')
plt.show()
model.summary()` 

[test2.zip](https://github.com/keras-team/keras/files/4528799/test2.zip)

**Comment by thisiseshan** [comment]
Hey I was able to make changes to your code and get the results. It seems there was an issue with the dimensions of **predict** and I noticed **weights** was initialised twice. I have used a different model here for faster processing. Test data would be found in **a** and corresponding predicted values in **b** Please let me know if this helped you. Cheers!

```x = []
y = []
for i in range(len(data)):
    #car = json.loads(data[i])
    car = data[i]
    if(car["Miles_per_Gallon"] is'None' or car["Miles_per_Gallon"] is None):
        continue
    if(car["Horsepower"] is None or car["Horsepower"] is 'null' or car["Cylinders"]  is None or car['Acceleration'] is None or car['Weight_in_lbs'] is None ):
        continue
    if(car["Horsepower"] != None or car["Horsepower"] != 'null' or car["Cylinders"] !=None or car["Cylinders"] != 'null' or car['Acceleration'] != 'null' or car['Weight_in_lbs'] != 'null' ):
        x.append([ car["Horsepower"]])
        y.append(car["Miles_per_Gallon"])

model = tf.keras.models.Sequential([
  tf.keras.layers.Dense(10, activation="relu", input_dim=1),
  tf.keras.layers.Dense(10, activation="relu"),
  tf.keras.layers.Dense(1)
])

optimizer = tf.keras.optimizers.SGD(lr=8e-6, momentum=0.9)
model.compile(loss="mse", optimizer=optimizer, metrics=['accuracy'])

model.fit(np.array(x), np.array(y),
          batch_size=int(len(x)*0.9),epochs=100,
          )

k = list()
z = list()

for i in range (10):
    t = randint(0,len(data))
    testData = x[i]
    testPredict = model.predict(testData)
    k.append(testData)
    z.append(testPredict)

a = np.array(k)
b = np.array(z) 

a = a.transpose()
b = b.T[0]```

**Comment by weiserhase** [comment]
i got the problem. One Dataset was not propperly serialized  but thanks with your awnser
